mava/components/building/best_checkpointer.py

```diff
@@ -30,19 +36,34 @@ def on_building_init(self, builder: SystemBuilder) -> None:
 
     def on_building_executor_start(self, builder: SystemBuilder) -> None:
         """Initialises the store for best model checkpointing"""
-        if not (builder.store.is_evaluator and self.config.checkpoint_best_perf):
+        if not (
+            builder.store.is_evaluator
+            and (self.config.checkpoint_best_perf or self.config.absolute_metric)
+        ):
             return
 
         builder.store.best_checkpoint = self.init_checkpointing_params(builder)
 
     def on_parameter_server_init(self, server: SystemParameterServer) -> None:
         """Adding checkpointing parameters to parameter server"""
-        if not self.config.checkpoint_best_perf:
-            return
-
-        server.store.parameters.update(
-            {"best_checkpoint": self.init_checkpointing_params(server)}
-        )
+        if self.config.checkpoint_best_perf:
+            server.store.parameters.update(
+                {"best_checkpoint": self.init_checkpointing_params(server)}
+            )
+
+        if self.config.absolute_metric:
+            if (
+                not (server.store.global_config.termination_condition is None)
+                and "executor_steps"
+                in server.store.global_config.termination_condition.keys()
+            ):
+                server.calculate_absolute_metric = True  # type: ignore
+            else:
+                logging.exception(
+                    f"{ValueError}: To calculate the absolute metric, you need to define\
+                    a termination condition related to the executor_steps."
```

Comment:
Won't it work with just any termination condition?

Reply:
I think only with the executor step because we need the number of steps to run the absolute metric, should we change it to be for any condition?

Reply:
In that case it is okay, since terminating by steps makes the most sense.